src/fromager/commands/build_order.py

```diff
@@ -298,12 +298,22 @@ def show_duplicates(build_order_file: pathlib.Path, install_only: bool):
     for name, steps in sorted(dependencies.items()):
         if len(steps) == 1:
             continue
-        print(f"{name}: building {len(steps)} versions")
+        print(f"\n{name}: building {len(steps)} versions")
         for step in steps:
             why = step["why"]
             req_type = why[-1][0]
+            if req_type[0] in "aeiou":
+                article = "an"
+            else:
+                article = "a"
             req = step["req"]
-            parents = ", ".join(w[1] for w in why[:-1])
-            print(
-                f"  '{req}' resolved to {step['version']} as a {req_type} dependency of {parents}"
-            )
+            print(f"  {req}")
+            if len(why) > 1:
+                parents = " -- ".join(f"'{w[1]}'" for w in why[:-1])
+                print(
+                    f"    resolved to {step['version']} as {article} {req_type} dependency of {parents}"
+                )
+            else:
+                print(
+                    f"    resolved to {step['version']} as {article} {req_type} dependency"
+                )
```

Comment:
```suggestion
            parents = " of " + " -- ".join(f"'{w[1]}'" for w in why[:-1]) if len(why) > 1 else ""
            print(f"\t resolved to {step['version']} as {article} {req_type} dependency{parents}")
```

Reply:
Do you find that easier to read? It seems more complex to me, especially with all of the quotes.

Reply:
fair i was trying to reduce the duplicate print statements. How about something like this?

```python
show_duplicate_msg = f"\t resolved to {step['version']} as {article} {req_type} dependency"
if len(why) > 1:
    parents = " -- ".join(f"'{w[1]}'" for w in why[:-1])
    show_duplicate_msg = f"{show_duplicate_msg} of {parents}"
print(show_duplicate_msg)
```

Reply:
That looks better, I'll make that update when I come back to this.

Reply:
Updated